components/clp-package-utils/clp_package_utils/scripts/native/search.py

```
@@ -297,10 +297,17 @@ def main(argv):
         return -1
 
     database_config: Database = clp_config.database
-    dataset = parsed_args.dataset
-    if dataset is not None:
+    datasets = parsed_args.dataset
+    if datasets is not None:
+        max_datasets_per_query = clp_config.query_scheduler.max_datasets_per_query
+        if max_datasets_per_query is not None and len(datasets) > max_datasets_per_query:
+            logger.error(
+                f"Number of datasets ({len(datasets)}) exceeds"
+                f" max_datasets_per_query={max_datasets_per_query}."
```

Ruff (0.15.1)

[warning] Logging statement uses f-string

(G004)